pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -71,6 +72,16 @@ const (
 )
 
 const (
+	cephOSDStart = `
+%s %s
+rc=$?
+if [ $rc -eq 0 ]; then
+	touch /tmp/osd-sleep
+	echo "OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 6 hrs. Restart the pod manually once the flapping issue is fixed"
+	sleep 6h
```

Comment:
As discussed, let's make the sleep time configurable. And if they want to disable it (equivalent of sleep time 0), perhaps we even consider keep the `ceph-osd` as the main process instead of starting as a script. This would give a way to effectively turn off the feature in case there is some issue with the bash script approach, although perhaps the risk isn't big enough to worry about that.

Reply:
Made this feature configurable from the CR spec.  Default value will be 6h.